### PR TITLE
v0.2.3: Add IDs to generated heading markup

### LINKED ISSUE #9: Headings from markdown don't have ids
Markdown library used in Jekyll adds ids to each heading, so

```### Something```

turns into

```<h3 id="something">Something</h3>```

This allows for deep linking and creating table of contents.

This was used quite heavily in Vanilla documentation and now doesn't work when we migrate it to Flask and templatefinder.

### PR DESCRIPTION
**Pull request by nottrobin**
Fixes #9

QA
--

Run this branch: https://github.com/canonical-web-and-design/vanilla-framework/pull/2792

Go to base/typography/ and see the in-page links work

### CODE REVIEW

File: canonicalwebteam/templatefinder/templatefinder.py [hunk 1]
```diff
@@ -25,6 +26,22 @@ class WebteamBlockLexer(BlockLexer):
     )
 
 
+class IDRenderer(Renderer):
+    def header(self, text, level, raw=None):
+        header_id = (text.replace(" ", "-")).lower()
+        header_id = re.sub("[^\w-]", "", header_id)
+        header_id = re.sub('<[^<]+?>', '', header_id)
```
Comment (bartaz) [inline canonicalwebteam/templatefinder/templatefinder.py]: Does it work here? If we want to remove `<...>` html tags shouldn't it be done before we remove special chars line before it?

Comment (nottrobin) [reply canonicalwebteam/templatefinder/templatefinder.py]: lol. it does indeed not work. and that is clearly why =)

**Comment by bartaz** [comment]
Seems like tests need updating (probably worth adding cases for headings with ids with special chars and html). And linter is complaining.

**Review by bartaz — APPROVED** [review]
LGTM

**Pull request merged**